Sources/SwiftDocC/Model/Rendering/References/ExternalLocationReference.swift

```diff
@@ -24,7 +24,16 @@ public struct ExternalLocationReference: RenderReference, URLReference {
 
     public private(set) var type: RenderReferenceType = .externalLocation
 
-    public var identifier: RenderReferenceIdentifier
+    public var identifier: RenderReferenceIdentifier {
+        didSet {
+            // Keep the url property in sync if it was just a copy of the reference identifier
+            if self.url == oldValue.identifier {
```

Comment:
I like this as a way to maintain the existing behavior but it does seem a little odd to me. Just from looking at the JSON spec – I don't think there's any implication that the URL is at all related to the identifier, it just happens to be in our current implementation.

Maybe we should have a `FIXME:` to remove this at some future point and explain in a comment that it's here for backwards compatibility? Or do you think this is something we should support longer-term? 


(To clarify a bit – it makes total sense that we set the `url` from the `identifier` in the `init` of this struct, I'm just not sure we should be providing the syncing behavior.)

Reply:
I added this to keep the existing behavior when you modify the `identifier` property between initialization and encoding. I took a guess that this was happening in tests somewhere (or in proper code), but i didn't actually investigate whether it was necessary. I'll take a look to see if that assumption holds.